packages/twenty-server/src/engine/core-modules/ask-ai/ask-ai.service.ts

```diff
@@ -0,0 +1,181 @@
+import { Injectable } from '@nestjs/common';
+
+import { SqlDatabase } from 'langchain/sql_db';
+import { RunnableSequence, RunnableFunc } from '@langchain/core/runnables';
+import { StringOutputParser } from '@langchain/core/output_parsers';
+import { isUUID } from 'class-validator';
+import { DataSource } from 'typeorm';
+import omit from 'lodash.omit';
+
+import { LLMPromptTemplateEnvVar } from 'src/engine/integrations/llm-prompt-template/interfaces/llm-prompt-template-name.interface';
+
+import {
+  AskAIQueryResult,
+  RecordMetadataById,
+} from 'src/engine/core-modules/ask-ai/dtos/ask-ai-query-result.dto';
+import { WorkspaceDataSourceService } from 'src/engine/workspace-datasource/workspace-datasource.service';
+import { WorkspaceQueryRunnerService } from 'src/engine/api/graphql/workspace-query-runner/workspace-query-runner.service';
+import { LLMPromptTemplateService } from 'src/engine/integrations/llm-prompt-template/llm-prompt-template.service';
+import { LLMChatModelService } from 'src/engine/integrations/llm-chat-model/llm-chat-model.service';
+import { LLMTracingService } from 'src/engine/integrations/llm-tracing/llm-tracing.service';
+import { ObjectMetadataService } from 'src/engine/metadata-modules/object-metadata/object-metadata.service';
+import { computeTableName } from 'src/engine/utils/compute-table-name.util';
+
+@Injectable()
+export class AskAIService {
+  constructor(
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+    private readonly workspaceQueryRunnerService: WorkspaceQueryRunnerService,
+    private readonly llmChatModelService: LLMChatModelService,
+    private readonly llmPromptTemplateService: LLMPromptTemplateService,
+    private readonly llmTracingService: LLMTracingService,
+    private readonly objectMetadataService: ObjectMetadataService,
+  ) {}
+
+  private async getRecordMetadataById(
```

Comment:
This is very dirty 😅. I don't think we can accept merging something like that, even temporary. It's a product question more than an eng question.
I'm not sure I understand the point entirely. Is it fetching relationships? it doesn't work for all records in my case.
<img width="867" alt="Screenshot 2024-06-23 at 11 19 06" src="https://github.com/twentyhq/twenty/assets/6399865/b56e4131-8a5e-431a-a786-5ee6040c725d">

I think we need to clarify the product scope of this which is really a DB viewer that is stronger on things like aggregates and joins but will not be strong on display ;  eventually a user should be able to view the list on records in a graphql/nice context, for example by redirecting to /objects?(id in X1...Xn) or something like that ; but he won't get the best best of both world. You cannot hope to have a raw sql query with joins and aggregates, and have a beautiful UI